src/uct/ib/mlx5/gga/gga_mlx5.c

```diff
@@ -275,9 +301,13 @@ static UCS_CLASS_INIT_FUNC(uct_gga_mlx5_ep_t, const uct_ep_params_t *params)
 
     UCS_CLASS_CALL_SUPER_INIT(uct_rc_mlx5_base_ep_t, params);
 
-    self->dma_opaque.mr = ibv_reg_mr(md->super.pd, self->dma_opaque.buf,
-                                     UCT_GGA_MLX5_OPAQUE_BUF_LEN,
-                                     IBV_ACCESS_LOCAL_WRITE);
+    self->dma_opaque.mr =
+            ibv_reg_mr(md->super.pd,
+                       ucs_align_up_pow2_ptr(&self->dma_opaque.buf[0],
+                                             UCS_SYS_CACHE_LINE_SIZE),
```

Comment:
yes, reg_mr < bufer size, it's needed for alignment

Reply:
won't it it better to allocate opaque_buf separately using posix_memalign for example?

Reply:
why is it better? the buffer is used on fast path, built-in buffer should improve memory locality, though I dont think performance impact can be proven by benchmarking...

Reply:
the code will be simpler IMO, less memory waste, and we don't really access the pointer contents so it should not matter. actually if hw is the only one accessing the pointer, it will be more efficient if it will be separated from the memory that the CPU is accessing to avoid false sharing